### ISSUE TITLE
How does multihead_attention works？

### ISSUE THREAD

**Issue by chenwi** [Kyubyong/transformer#49]

**Comment by chenwi** [comment]
`# Linear projections
        Q = tf.layers.dense(queries, num_units, activation=tf.nn.relu)  # (N, T_q, C)
        K = tf.layers.dense(keys, num_units, activation=tf.nn.relu)  # (N, T_k, C)
        V = tf.layers.dense(keys, num_units, activation=tf.nn.relu)  # (N, T_k, C)`
It seems that these codes share the  same Wi

**Comment by sunnnnnnnny** [comment]
@chenwi  No, the weights of the three full connections are learned by the network itself.
